acerhk, sched: Clean up preempt_enable_no_resched() abuse

The only valid use of preempt_enable_no_resched() is if the very next
line is schedule() or if we know preemption cannot actually be enabled
by that statement due to known more preempt_count 'refs'.

Signed-off-by: Stefan Lippers-Hollmann <s.l-h@gmx.de>

--- a/drivers/platform/x86/acerhk.c
+++ b/drivers/platform/x86/acerhk.c
@@ -181,7 +181,7 @@ static void send_kbd_cmd(unsigned char c
 
 	if (!my_i8042_wait_write())
 		outb(val, KBD_DATA_REG);
-	preempt_enable_no_resched();
+	preempt_enable();
 }
 
 static void enable_wlan_ec_1(void)
@@ -270,7 +270,7 @@ static void enable_mail_led_ec_3(void)
 
 	if (!my_i8042_wait_write())
 		outw(0xC061, 0xCFC);
-	preempt_enable_no_resched();
+	preempt_enable();
 }
 
 static void disable_mail_led_ec_3(void)
@@ -281,7 +281,7 @@ static void disable_mail_led_ec_3(void)
 
 	if (!my_i8042_wait_write())
 		outw(0xC060, 0xCFC);
-	preempt_enable_no_resched();
+	preempt_enable();
 }
 /* Experimental use of dritek keyboard extension }}} */
 
@@ -342,7 +342,7 @@ static int get_fnkey_event(void)
 
 	preempt_disable();
 	call_bios(&regs);
-	preempt_enable_no_resched();
+	preempt_enable();
 
 	return regs.eax & 0xffff;
 }
@@ -362,7 +362,7 @@ static int get_thermal_event(void)
 
 		preempt_disable();
 		call_bios(&regs);
-		preempt_enable_no_resched();
+		preempt_enable();
 
 		if (verbose >= 4)
 			pr_info("thermal event = 0x%x\n", regs.eax);
@@ -395,7 +395,7 @@ static void wbutton_fct_1(int val)
 
 		preempt_disable();
 		call_bios(&regs);
-		preempt_enable_no_resched();
+		preempt_enable();
 
 		if (verbose >= 4)
 			pr_info("wbutton1 = 0x%x\n", regs.eax);
@@ -423,7 +423,7 @@ static void wbutton_fct_2(int val)
 
 		preempt_disable();
 		call_bios(&regs);
-		preempt_enable_no_resched();
+		preempt_enable();
 
 		if (verbose >= 4)
 			pr_info("wbutton2 = 0x%x\n", regs.eax);
@@ -474,7 +474,7 @@ static void set_mail_led(int val)
 
 		preempt_disable();
 		call_bios(&regs);
-		preempt_enable_no_resched();
+		preempt_enable();
 
 		if (verbose >= 4)
 			pr_info("mail led set to = 0x%x\n", val);
@@ -514,7 +514,7 @@ static void launch_connect(int val)
 
 		preempt_disable();
 		call_bios(&regs);
-		preempt_enable_no_resched();
+		preempt_enable();
 
 		if (verbose >= 4)
 			pr_info("connect(%d) = 0x%x\n", val, regs.eax);
